python/hsfs/engine/python.py

```diff
@@ -91,14 +91,24 @@ def sql(
         schema=None,
     ):
         if not online_conn:
-            return self._sql_offline(sql_query, feature_store, dataframe_type, schema)
+            return self._sql_offline(
+                sql_query,
+                feature_store,
+                dataframe_type,
+                schema,
+                hive_config=read_options.get("hive_config") if read_options else None,
```

Comment:
Isn't it always going to be an empty dict in the worst case?

Then you could do
```suggestion
                hive_config=read_options.get("hive_config", None),
```

Reply:
`read_options` can be None.